src/pages/Authentication/Signup.jsx

```diff
@@ -0,0 +1,146 @@
+import "./authentication.css";
+import { Link, useNavigate, useLocation } from "react-router-dom";
+import { useState } from "react";
+import { signupService } from "../../services/authServices";
+import { useAuth } from "../../context/userContext";
+import { Loader } from "../../Components";
+
+const Signup = () => {
+  const {
+    authState: { loading },
+    authDispatch,
+  } = useAuth();
+  const [userDetails, setUserDetails] = useState({
+    fullName: "",
+    phNumber: "",
+    email: "",
+    password: "",
+  });
+  const [showPassword, setShowPassword] = useState(false);
+  const [errorMsg, setErrorMsg] = useState("");
+  const navigate = useNavigate();
+  const location = useLocation();
+
+  const signupHandler = async (e) => {
+    e.preventDefault();
+    const { fullName, phNumber, email, password } = userDetails;
+    if (fullName && phNumber && email && password) {
+      const response = await signupService(
+        { fullName, phNumber, email, password },
+        authDispatch
+      );
+      if (response.hasOwnProperty("errorMsg")) {
+        setErrorMsg(response.errorMsg);
+      } else {
+        setUserDetails({
+          ...userDetails,
+          fullName: "",
+          phNumber: "",
+          email: "",
+          password: "",
+        });
+        navigate(location.state?.from?.pathname || "/", { replace: true });
+        setErrorMsg("");
+      }
+    } else {
+      setErrorMsg("Enter all the details");
+    }
+  };
+
+  return (
+    <div className="auth-main">
+      {loading && <Loader />}
+      {!loading && (
+        <div className="auth-container">
+          <h3 className="txt-center txt-lg form-heading">Signup</h3>
+          <form className="txt-form">
+            <label htmlFor="name" className="txt-md form-label">
+              Name
+            </label>
+            <input
```

Comment:
suggestion :- use implicit label (wrap input around label)

Reply:
Thankyou for your suggestion will implement this in next pr